### ISSUE TITLE
Results of the semantic highlighter aren't properly displayed in VS Code

### ISSUE THREAD

**Issue by sungshik** [usethesource/rascal-language-servers#458]
The purpose of opening this issue is to document a bug that I haven't been able to reproduce after observing it once. Others, though, have also reported occasionally observing similar strange behavior, but it hasn't happened often enough to figure out the conditions that trigger it. Also, there's reason to believe that this is actually a bug in VS Code (maybe a race of some sort).

Consider the following screenshots:

![screen1](https://github.com/user-attachments/assets/b5bd74c2-a508-45e8-96c9-3a2c37f7c778)
![screen2](https://github.com/user-attachments/assets/b8120672-a899-4f92-bce3-ec8c67212cac)
![screen3](https://github.com/user-attachments/assets/beec807d-f971-4b94-9248-7be05bc0b49f)

The screenshots differ in the position of the cursor: `|3.4` (screenshot 1), `3|.4` (screenshot 2), and `3.|4` (screenshot 3). The scope inspector in the screenshots shows that:
  - the current token is `3.14`;
  - the foreground color of the current token is `#B5CEA8`.

However, the actual color of `3.14` in the editor is `#B5CEA8` for `3` but `#D4D4D4` for `.4`. This is confirmed by inspecting the HTML:

![image](https://github.com/user-attachments/assets/f148326c-ea1e-407a-bdc2-b47902474606)
![image](https://github.com/user-attachments/assets/5e1575cd-8815-491d-b6e1-264c6db4246f)

Thus, **the following are inconsistent:**
  - the metadata about the token in the scope selection (say, the model);
  - the actual color of the token in the HTML (say, the view).

This doesn't seem like something that (the semantic tokenizer in) our LSP server should be able to cause.

Just after making these screenshots, it turned out that the bug manifests only when the file *doesn't* end with a newline:

![image](https://github.com/user-attachments/assets/060d640a-b4e5-43d3-a267-443d9b658ddf)

However, since then, I haven't been able to reproduce the bug. This makes it a bit hard to further diagnose and/or file a bug report with VS Code. We'll just wait for now until the bug resurfaces.

**Comment by jurgenvinju** [comment]
Yes I've seen this too way back when and was hoping it would go away 😇 it doesn't happen that often.
